octobot_trading/personal_data/positions/position.py

```diff
@@ -336,27 +352,42 @@ def _is_update_closing(self, size_update):
             return self.size + size_update <= constants.ZERO
         return self.size + size_update >= constants.ZERO
 
-    def _update_size(self, update_size):
+    def _update_size(self, size_update, closed_order=None, update_order_margin=False):
         """
         Updates position size and triggers size related attributes update
-        :param update_size: the size quantity
+        :param size_update: the size quantity
         :return: True if the update increased position size
         """
-        is_update_increasing_position_size = self._is_update_increasing_size(update_size)
-        if not is_update_increasing_position_size:
-            self._update_realized_pnl_from_size_update(update_size, is_closing=self._is_update_closing(update_size))
-        self._check_and_update_size(update_size)
+        order_price = constants.ZERO if closed_order is None else closed_order.filled_price
+        is_increasing_size = self._is_update_increasing_size(size_update)
+        realized_pnl_update = constants.ZERO
+        if self._is_update_decreasing_size(size_update):
+            realized_pnl_update = self._update_realized_pnl_from_size_update(
+                size_update,
+                is_closing=self._is_update_closing(size_update),
+                order_price=order_price
+            )
+            # TODO check: funding fees when no wallet balance: as they reduce the size of the position, the final
+            # realized pnl will be smaller, it should therefore be taken into account.
+        self._check_and_update_size(size_update)
         self._update_quantity()
         self._update_side()
         if self.exchange_manager.is_simulated:
-            self._update_initial_margin()
+            margin_update = self._update_initial_margin()
+            if closed_order is None:
+                self.on_margin_update(margin_update, update_order_margin=update_order_margin)
+            else:
+                # TODO: check if needed in real trader ?
```

Comment:
How could we need it?

Reply:
Here is a question for you :)
=> are there elements in position/portfolio that we need to update in real trading or can we just rely on exchange data for every aspect of it ?